fix: bug when building search index

n05la3 opened the pull request:
When building the search indices (`yarn build`), you get an error that says file `docs.md`, `grid.md` could not be found. This happens because the docs page corresponds to `src/Homepage.vue` (is a NOT a `.md` file) and unlike the other `.md` pages is not found in the location the parser would expect. The parser which builds the search index expects to find a markdown file called `docs.md` in the pages folder and similarly, it expects to find `grid.md` in `pages/vue-components/grid.md` which is rather found in `pages/ComponentsPage.vue`. 

This PR handles these edge cases.

n05la3 committed:
fix: fix bug when building search index, ensure custom paths are properly handled

IlCallo commented:
Can you explain what was the problem? Mainly for documentation purposes

n05la3 commented:
Just updated the description.

n05la3 committed:
fix(routes): import roadmap from web instead of local, importing from local throws an error as vite can't find vue dependency

IlCallo commented on docs/src/router/routes.js:
```diff
@@ -27,7 +27,8 @@ const docsPages = [
     },
-    component: () => import('../../../ROADMAP.md')
+    // importing from ../../../ROADMAP.md throws an error as vite is unable to find vue dependency
+    component: () => import('https://roadmap.quasar.dev/')
```
This isn't what I meant by "I'd redirect start/roadmap to https://roadmap.quasar.dev/ directly" 👀 
Have you tested this change by running the docs in dev mode? And if yes, by building docs and running them via `quasar serve`? Cause it doesn't work in both ways

What I meant was "make the ROADMAP button a link to the external `https://roadmap.quasar.dev/` URL", unless you find a way to make it play cool with Vite and keep using the MD file as a single source of truth

The import you changed doesn't solve the compilation problem, it moves it at runtime 😅 
Always test stuff before committing

n05la3 committed:
fix(docs/roadmap): prefer redirecting roadmap to https://roadmap.quasar.dev/ to avoid vite errors router
- One issue with this method is that roadmap related searches will no longer appear when searching, as indices.json depends on markdown files for creating filters

n05la3 commented:
One other thing about redirecting to `https://roadmap.quasar.dev/` is that `roadmap` related searches will no longer appear in the search dropdown. Until we find a better fix for managing the vite issue (it's unable to find vue dependency when reading `ROADMAP.md` from docs root), I don't think this is a big issue as `roadmap` is easily accessible from the header.
